File: app/src/main/java/sp/inetvpn/ui/ContactActivity.java
package sp.inetvpn.ui;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;

import sp.inetvpn.R;
import sp.inetvpn.databinding.ActivityContactBinding;
import sp.inetvpn.handler.SendFeedback;
import sp.inetvpn.util.CheckInternetConnection;

public class ContactActivity extends Activity {
    private ActivityContactBinding binding;

    String feedback, more, email;
    String Connecting, Crashed, Servers;

    @Override
    public void onBackPressed() {
        finish();
        overridePendingTransition(R.anim.anim_slide_in_left, R.anim.anim_slide_out_right);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityContactBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.btnAboutContactSubmit.setOnClickListener(v -> {
            // advertising
            if (CheckInternetConnection.netCheck(this)) {

                if (binding.checkboxAboutContactConnecting.isChecked()) {
                    Connecting = "true";
                } else {
                    Connecting = "false";
                }

                // speed
                if (binding.checkboxAboutContactCrashed.isChecked()) {
                    Crashed = "true";
                } else {
                    Crashed = "false";
                }

                // connecting
                if (binding.checkboxAboutContactServers.isChecked()) {
                    Servers = "true";
                } else {
                    Servers = "false";
                }

                Bundle paramsFeed = new Bundle();
                paramsFeed.putString("Connecting", Connecting);
                paramsFeed.putString("Crashed", Crashed);
                paramsFeed.putString("Servers", Servers);

                feedback = String.valueOf(paramsFeed);
                more = binding.etAboutContactOtherProblems.getText().toString();
                email = binding.etAboutContactEmail.getText().toString();

                SendContactLog Object = new SendContactLog();
                Object.start();

                binding.btnAboutContactSubmit.setText("ارسال شد");
                binding.btnAboutContactSubmit.setEnabled(false);

            }
        });

        LinearLayout ll_about_contact_close = findViewById(R.id.ll_contact_back);
        ll_about_contact_close.setOnClickListener(v -> {
            finish();
            overridePendingTransition(R.anim.anim_slide_in_left, R.anim.anim_slide_out_right);
        });

    }

    class SendContactLog extends Thread {
        @Override
        public void run() {
            SendFeedback.sendFeedBack(ContactActivity.this, feedback, more, email);
        }

    }

}

File: app/src/main/java/sp/inetvpn/ui/ContactActivity.kt
package sp.inetvpn.ui

import android.app.Activity
import android.os.Bundle
import android.view.View
import sp.inetvpn.R
import sp.inetvpn.databinding.ActivityContactBinding
import sp.inetvpn.handler.SendFeedback
import sp.inetvpn.util.CheckInternetConnection

class ContactActivity : Activity() {
    private var binding: ActivityContactBinding? = null
    var feedback: String? = null
    var more: String? = null
    var email: String? = null
    private var connecting: String? = null
    private var crashed: String? = null
    private var servers: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityContactBinding.inflate(layoutInflater)
        val view: View = binding!!.getRoot()
        setContentView(view)
        binding!!.btnAboutContactSubmit.setOnClickListener {
            // advertising
            if (CheckInternetConnection.netCheck(this)) {
                connecting = if (binding!!.checkboxAboutContactConnecting.isChecked) {
                    "true"
                } else {
                    "false"
                }

                // speed
                crashed = if (binding!!.checkboxAboutContactCrashed.isChecked) {
                    "true"
                } else {
                    "false"
                }

                // connecting
                servers = if (binding!!.checkboxAboutContactServers.isChecked) {
                    "true"
                } else {
                    "false"
                }
                val paramsFeed = Bundle()
                paramsFeed.putString("Connecting", connecting)
                paramsFeed.putString("Crashed", crashed)
                paramsFeed.putString("Servers", servers)
                feedback = paramsFeed.toString()
                more = binding!!.etAboutContactOtherProblems.getText().toString()
                email = binding!!.etAboutContactEmail.getText().toString()

                SendContactLog().start()
                binding!!.btnAboutContactSubmit.text = "ارسال شد"
                binding!!.btnAboutContactSubmit.setEnabled(false)
            }
        }
        binding!!.headerLayout.llBack.setOnClickListener { v: View? -> onBackPressed() }
    }

    internal inner class SendContactLog : Thread() {
        override fun run() {
            SendFeedback.sendFeedBack(this@ContactActivity, feedback, more, email)
        }
    }

    @Deprecated("Deprecated in Java")
    override fun onBackPressed() {
        finish()
        overridePendingTransition(R.anim.anim_slide_in_left, R.anim.anim_slide_out_right)
    }
}
